brokers/auth/mode.go

```diff
@@ -0,0 +1,83 @@
+package auth
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ubuntu/authd/brokers/layouts"
+	"github.com/ubuntu/authd/internal/proto/authd"
+)
+
+// Mode is the type to define an authd authentication Mode for brokers usage.
+type Mode struct {
+	*authd.AuthMode
```

Comment:
I don't like that this adds a dependency on protobuf and grpc to the broker (which doesn't actually use gRPC). I think we need to redefine the fields instead of embedding the protobuf message here. 

I also think we can get rid of most of the boilerplate here by using JSON as an intermediate format when converting to/from protobuf and string map. I have some WIP to implement that but some more work is needed and I cannot in good conscience invest more time in it now (while I have other, scheduled tasks to work on). Do you think this could wait until next pulse?

Reply:
This was my concern too... But I wanted to get it done quickly and this allowed both API consistency and generated code.

So I had also the idea of just reusing it fully privately, but it would still likely lead to the same dependency problem.

Thus, I can definitely refactor this too (I had already thought about, then I decided to see how the shortcut was accepted, at least as an interim solution).

Not sure about passing from json though. It was also one of the reasons why I was using the authd protobufs types (as json conversion is ready and works great as we do for gdm), but I wanted to avoid doing native structs -> variant -> native map -> json -> native structs.

Even though I had it in my options.

One thing I can probably craft is instead a simple generator that reimplements the protobuf structures into something proto-less.

Reply:
> Not sure about passing from json though. It was also one of the reasons why I was using the authd protobufs types (as json conversion is ready and works great as we do for gdm), but I wanted to avoid doing native structs -> variant -> native map -> json -> native structs.

I assume you're worried about the performance cost, or is there another reason you want to avoid that?

Reply:
I suspect that the overhead doesn't make much difference in practice, and that the advantage of not having to maintain this much boilerplate outweighs the performance impact. Maybe we can discuss this further in a call (next week or whenever we have time for this issue), then I could show you the WIP I have.

Reply:
Ok, let's handle that in a followup, I've removed this part from the PR for now.

Reply:
> Ok, let's handle that in a followup

That's https://warthogs.atlassian.net/browse/UDENG-5522